template/src/utils/useGroupMessages.ts

```diff
@@ -0,0 +1,24 @@
+/*
+********************************************
+ Copyright © 2021 Agora Lab, Inc., all rights reserved.
+ AppBuilder and all associated components, source code, APIs, services, and documentation 
+ (the “Materials”) are owned by Agora Lab, Inc. and its licensors. The Materials may not be 
+ accessed, used, modified, or distributed for any purpose without a license from Agora Lab, Inc.  
+ Use without a license or in violation of any license terms and conditions (including use for 
+ any purpose competitive to Agora Lab, Inc.’s business) is strictly prohibited. For more 
+ information visit https://appbuilder.agora.io. 
+*********************************************
+*/
+import {useContext} from 'react';
+import ChatContext from '../../src/components/ChatContext';
+
+/**
+ * This hook will fetch the group messages
+ * @returns userList
+ */
+function useGroupMessages() {
+  const {messageStore} = useContext(ChatContext);
```

Comment:
With all use of useContext like this, wrap it within a conditional, do the same for usePrivateMessage as well
```suggestion
  const context = useContext(ChatContext);
  if (context === undefined) {
    throw new Error(`useGroupMessages must be used within a ChatContext`);
  }
  return context;

```

Reply:
useContext will return the default vaule. so we can't identify whether is wrapped with the provider or not. we will discuss on this one